### PR TITLE
Instrumentation: Measure app init load times

### PR DESCRIPTION
**Pull request by mckn**
**What is this feature?**

In this PR we are starting to send metrics of how much time it takes to load `appInit` in Grafana. We also send separate metrics for the preload plugins section and each preloaded plugin to make it possible to measure those individually.

Screenshot 2023-05-05 at 12 24 37

**Why do we need this feature?**

When we added the UI extension feature it, currently, require the plugin to preload. We need to keep track of the performance impact when more and more plugins will start to use this feature (being preloaded).

**Who is this feature for?**

Grafana developers

**Which issue(s) does this PR fix?**:

None

**Special notes for your reviewer:**

I'm kind of new to the `performance` API so please have a look at how we are using it and if it can be done in a better way.

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

**Review by torkelo — COMMENTED** [review]
looks good!

**Review by leventebalogh — APPROVED** [review]
Left one commend regarding the location of the new module, but overall LGTM 👍

File: public/app/metrics.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    const measure = performance.measure(measured, started, completed);
+    reportPerformance(`${eventName}_ms`, measure.duration);
+
+    performance.clearMarks(started);
+    performance.clearMarks(completed);
+    performance.clearMeasures(measured);
+  } catch (error) {
+    console.error(`[Metrics] Failed to stopMeasure ${eventName}`, error);
+  }
+}
```
Comment (leventebalogh) [inline public/app/metrics.ts:35]: Should this maybe live under `public/core/utils/metrics.ts`? 
(Sounds like a utility that can be used in multiple places of Grafana core as a utility)

**Comment by leventebalogh** [comment]
Created an experimental PR to catch events that were sent in before the `EchoSrv` was initialised: https://github.com/grafana/grafana/pull/67977 (With that we could probably remove the warning comment in the PR?)

**Review by jackw — APPROVED** [review]
LGTM! 🚀 

+1 with @leventebalogh suggestion of putting it in `core/utils`.

**Pull request merged**